keps/sig-scheduling/4815-dra-partitionable-devices/README.md

```diff
@@ -252,222 +252,465 @@ non-goals of this KEP.
+  // their name. All others must be listed with their domain prefix.
+  //
+  // Conflicting capacities from those provided via other mixins are
+  // overwritten by the ones provided here.
+  //
+  // The maximum number of attributes and capacities combined is 32.
+  //
+  // +optional
+  Capacity map[QualifiedName]DeviceCapacity
+
+  // ConsumesCapacity defines a list of references to capacity
+  // pools and the set of capacities that the device will
+  // consume from those pools.
+  //
+  // The capacities can be defined either by referencing one
+  // or more DeviceCapacityConsumptionMixins for by listing
+  // the capacities directly. The latter will always override
+  // any capacities coming in from the mixins.
+  //
+  // The maximum number of device capacity consumption entries
+  // is 32. This is the same as the maximum number of capacity
+  // pools allowed in a ResourceSlice.
+  //
+  // +required
+  // +listType=atomic
+  ConsumesCapacity []DeviceCapacityConsumption
+}
+
+// CapacityPoolMixin defines a mixin that a capacity pool can include.
+type CapacityPoolMixin struct {
+  // Name is a unique identifier among all capacity pool mixins in the ResourceSlice.
+  // It must be a DNS label.
+  //
+  // +required
+  Name string
+
+  // Capacity defines the set of capacities for this mixin.
+  // The name of each capacity must be unique in that set.
+  //
+  // To ensure this uniqueness, capacities defined by the vendor
+  // must be listed without the driver name as domain prefix in
+  // their name. All others must be listed with their domain prefix.
+  //
+  // Conflicting capacities from those provided via other mixins are
+  // overwritten by the ones provided here.
+  //
+  // The maximum number of capacities is 32.
+  //
+  // +required
+  Capacity map[QualifiedName]DeviceCapacity
+}
+
+// CapacityPool defines a named pool of capacities
+// that are available to be used by devices defined in the
+// ResourceSlice.
+//
+// The capacities are not allocatable by themselves, but
+// can be referenced by devices. When a device is allocated,
+// the capacity it uses will no longer be available for use
+// by other devices.
+type CapacityPool struct {
+  // Name defines the name of the capacity pool.
+  // It must be a DNS label.
+  //
+  // +required
+  Name string
+
+  // Includes defines the set of capacity pool mixins that this capacity
+  // pool includes.
+  //
+  // The propertes of each included mixin are applied to this capacity pool in
+  // order. Conflicting properties from multiple mixins are taken from the
+  // last mixin listed that contains them. Properties set on the capacity pool will
+  // always override properties from mixins.
+  //
+  // The mixins referenced here must be defined in the same
+  // ResourceSlice.
+  //
+  // The maximum number of mixins that can be included is 8.
+  //
+  // +optional
+  // +listType=atomic
+  Includes []CapacityPoolMixinRef
```

Comment:
@klueska Is there a particular reason why the original design has references as a list of objects with just the `Name` field rather than just a list of strings? I've kept all the `<Something>Ref` types consistent with this approach, but using a separate struct is a little bit more verbose than just a list of strings.

Reply:
It was just to leave it open to extension in the future. We do the same for the reference to rsourceclaims in a pod spec, for example.

Reply:
Yeah, that makes sense.